BUILD.gn

```
@@ -1189,6 +1198,49 @@ if (is_mac) {
   }
 }
 
+test("shell_browser_ui_unittests") {
+  sources = [
+    "//electron/shell/browser/ui/accelerator_util_unittests.cc",
+    "//electron/shell/browser/ui/run_all_unittests.cc",
+  ]
+
+  configs += [ ":electron_lib_config" ]
+
+  deps = [
+    ":electron_lib",
+    "//base",
+    "//base/test:test_support",
+    "//testing/gmock",
+    "//testing/gtest",
+    "//ui/base",
+    "//ui/strings",
+  ]
+
+  if (is_mac) {
+    libs = [
+      "AVFoundation.framework",
+      "Carbon.framework",
+      "LocalAuthentication.framework",
+      "QuartzCore.framework",
+      "Quartz.framework",
+      "Security.framework",
+      "SecurityInterface.framework",
+      "ServiceManagement.framework",
+      "StoreKit.framework",
+    ]
+
+    # Resolve paths owing to different test executable locations
+    ldflags = [
+      "-F",
+      rebase_path("external_binaries", root_build_dir),
+      "-rpath",
+      "@loader_path",
+      "-rpath",
+      "@executable_path/Electron.app/Contents/Frameworks",
```

This is the wrong rpath to use here, because it depends on the app having been built. `Electron.app/Contents/Frameworks` contains symlinks to the "real" files, so instead, we should use the paths to the directory(s) containing real frameworks directly here.

---

i'm not quite sure what you mean here? i don't see the custom (non-system, like Squirrel) frameworks existing anywhere else 🤔

---

Those frameworks get assembled into the `.app` bundle as part of the `electron_app_framework_bundle_data` target, so if that target doesn't get built (e.g. if you build `shell_browser_ui_unittests` without having built `electron_app`) then those frameworks won't have been copied there.

Maybe something like:
```suggestion
      rebase_path("external_binaries", root_build_dir),
```

? That will produce a binary with an absolute-path link command, which isn't great, but the test binary isn't likely to get run from anywhere but the exact place it was built, so it's probably fine?

---

hmm; that's what's already at the top in the existing PR, right? https://github.com/electron/electron/pull/20293/files#diff-4eaebbcd1dc0da11e3a3ce488290bfc6R1224

That doesn't work for the above reasons; it fails to find all local frameworks.

---

`-F` and `rpath` are different flags that do different things. `-F` is where to find frameworks at build-time, `rpath` is where to find them at run-time.